src/pudl/settings.py

```diff
@@ -243,41 +244,71 @@ class Eia860Settings(GenericDatasetSettings):
         data_source: DataSource metadata object
         years: list of years to validate.
         eia860m: whether or not to incorporate an EIA-860m month.
-        eia860m_year_month ClassVar[str]: The 860m year-month to incorporate.
+        eia860m_year_months ClassVar[str]: The 860m year-month to incorporate.
     """
 
     data_source: ClassVar[DataSource] = DataSource.from_id("eia860")
     years: list[int] = data_source.working_partitions["years"]
-
     eia860m: bool = True
-    eia860m_year_month: ClassVar[str] = max(
-        DataSource.from_id("eia860m").working_partitions["year_months"]
-    )
+    all_eia860m_year_months: list[str] = DataSource.from_id(
+        "eia860m"
+    ).working_partitions["year_months"]
```

Comment:
I tried making `all_eia860m_year_months` a private variable by adding an `_` in front (because it shouldn't ever be changed/altered/used as anything but a reference within the class functions). However, when I added the underscore, I could no longer reference it in the class functions (ex: `info.data["_all_eia860m_year_months]`). So for now I removed the `_`. If anyone knows how to do this successfully lmk!

Reply:
Our Pydantic settings classes are immutable, meaning that none of their attributes can be changed once the class has been instantiated. This is something Pydantic implements, not Python (Python doesn't natively let you make class attributes private, really). I suspect that Pydantic is taking the "private" hint from the leading underscore and purposefully not making that attribute available in the dictionary of values in the validator.